lib/puppet/provider/cisco_aaa_group_tacacs/nxapi.rb

```diff
@@ -101,59 +111,32 @@ def properties_set(new_aaa_group=false)
   end
   # rubocop:enable GuardClause
 
-  # can't autogen getters and setters because the default_<prop>
-  # functions are class functions
-  def deadtime
-    return :default if @resource[:deadtime] == :default &&
-                       @property_hash[:deadtime] ==
-                       Cisco::AaaServerGroup.default_deadtime
-    @property_hash[:deadtime]
-  end
-
-  def deadtime=(set_value)
-    set_value = Cisco::AaaServerGroup.default_deadtime if
-      set_value == :default
-    @property_flush[:deadtime] = set_value
-  end
-
+  # can't autogen server_hosts, special array handling
   def server_hosts
     return [:default] if @resource[:server_hosts] &&
                          @resource[:server_hosts][0] == :default &&
                          @property_hash[:server_hosts] ==
-                         Cisco::AaaServerGroup.default_servers
+                         @aaa_group.default_servers
```

Comment:
nit: this would be cleaner if you do:

```
    return [:default] if
      @resource[:server_hosts] &&
      @resource[:server_hosts][0] == :default &&
      @property_hash[:server_hosts] == @aaa_group.default_servers
```

your call